### PR TITLE
chore(component-testing): CT-140 start TS migration

### PR DESCRIPTION
**Pull request by lmiller1990**
Far from perfectly typed, but good enough for now

### CODE REVIEW

File: packages/server-ct/src/server-ct.ts [hunk 1]
```diff
@@ -104,14 +107,10 @@ class Server {
...
 
-  _port () {
-    return _.chain(this._server).invoke('address').get('port').value()
-  }
-
   _listen (port, onError) {
     return new Bluebird((resolve) => {
       const listener = () => {
-        const address = this._server.address()
+        const address = this._server.address() as AddressInfo
```
Comment (JessicaSachs) [inline packages/server-ct/src/server-ct.ts:113]: Bart and I ran into this while pairing in the webpack-devserver. `address()` can also come back as a string. Is it worth handling this use case?

File: packages/server-ct/src/server-ct.ts [hunk 2]
```diff
@@ -126,36 +125,37 @@ class Server {
...
+  }
+
   _close () {
     // bail early we dont have a server or we're not
     // currently listening
     if (!this._server || !this.isListening) {
       return Bluebird.resolve()
     }
 
-    return this._server.destroyAsync()
```
Comment (JessicaSachs) [inline packages/server-ct/src/server-ct.ts:146]: Why are we no longer returning the value of this? Was it void?

File: packages/server-ct/src/server-ct.ts [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
-const debug = require('debug')('cypress:server-ct:server')
-const express = require('express')
-const Bluebird = require('bluebird')
-
-const allowDestroy = require('@packages/server/lib/util/server_destroy')
-const templateEngine = require('@packages/server/lib/template_engine')
-const Socket = require('./socket-ct')
+import http from 'http'
+import { promisify } from 'util'
+import _debug from 'debug'
```
Comment (JessicaSachs) [inline packages/server-ct/src/server-ct.ts:3]: I've been preferring `import debug as debugFn from 'debug'` inside of typescript -- but that's not a hard and fast convention

**Review by JessicaSachs — COMMENTED** [review]
Some questions. Other than that, looks good!

**Review by elevatebart — APPROVED** [review]

**Comment by elevatebart** [comment]
The chai typing issue is source of concern.
The rests seems good

File: packages/server-ct/src/server-ct.ts (hunk 1 above)
Comment (lmiller1990) [reply packages/server-ct/src/server-ct.ts:113]: ohhh snap, yes it is

File: packages/server-ct/src/server-ct.ts (hunk 3 above)
Comment (lmiller1990) [reply packages/server-ct/src/server-ct.ts:3]: I dont think `as` is valid for a default export

File: packages/server-ct/src/server-ct.ts (hunk 2 above)
Comment (lmiller1990) [reply packages/server-ct/src/server-ct.ts:146]: seems `_destroyAsync` never existed in the first place - but I added it (copy pasting) so I will put this back in

**Pull request merged**